cime/scripts/tests/scripts_regression_tests.py

```diff
@@ -152,6 +152,17 @@ def test_lib_doctests(self):
                         run_cmd_assert_result(self, 'PYTHONPATH={}:$PYTHONPATH python -m doctest {} 2>&1'.format(LIB_DIR, filepath), from_dir=LIB_DIR)
                     else:
                         print("{} has no doctests".format(filepath))
+###############################################################################
+class C_TestGridGeneration(unittest.TestCase):
+###############################################################################
+    def test_gen_domain(self):
+        inputdata = MACHINE.get_value("DIN_LOC_ROOT")
+        cime_root = get_cime_root()
+        testname = os.path.join(cime_root, "tools", "mapping", 
+                                "gen_domain_files", "test_gen_domain.sh")
+        temp_dir = tempfile.mkdtemp()
```

Comment:
Minor point: generally unittests should not do their cleanup directly in the test. It should be done in a teardown routine.